[LTP]: report results as junit

woju opened the pull request:
This is interconnected with oscarlab/graphene#896. This is largely untested, because I don't have another Jenkins instance. Expect several iterations.

closes #7 

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene-tests/21)
<!-- Reviewable:end -->

woju committed:
[LTP] report results as junit

yamahata reviewed (dismissed):
I haven't tested this myself.
As long as CI is happy, it looks good to me.

> Reviewed 15 of 15 files at r1.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-:-LlYWbLN8hC9ADMfTqiw:b-ucvam5)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @woju)

___
*[ltp/.gitignore, line 4 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LlYShJ_0XYyE51yHjo5:-LlYShJ_0XYyE51yHjo6:b-gv15vt) ([raw file](https://github.com/oscarlab/graphene-tests/blob/4a5426defa82a2d36c191c33f01dc83c8ea50455/ltp/.gitignore#L4)):*
> ```
> /Broken.csv
> /Failed.csv
> /Passed.csv
> ```

Because this PR removes block_tests.awk etc, those files won't be created any more, correct?
___
*[ltp/runltp_xml.py, line 71 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LlYW0oV3GtwIxntSQ5M:-LlYW0oW1qQQA4ScS-g-:b-p2z298) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L71)):*
> ```Python
>             return super().get(self.default_section, *args, **kwds)
> 
> class TestRunner:
> ```

Oh you wrote your own test runner and result formatter.
Isn't there any suitable existing one?
Just I'm curious.  This implementation is not so long, so I'm fine with this.


<!-- Sent from Reviewable.io -->

mkow reviewed (changes requested):
> Reviewed 15 of 15 files at r1.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @woju)

___
*[ltp/ltp.conf, line 441 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_D5fVB4z09G7gg0DE:-Ll_D5fW-24h9udrirqL:bsmc50x) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/ltp.conf#L441)):*
> ```
> skip = yes
> 
> # vim: ft=dosini
> ```

Can we name this file `ltp_conf.ini` instead of using modelines? Using a proper extension would be more portable.
___
*[ltp/README, line 4 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_DpxeEAR6irC401fv:-Ll_DpxeEAR6irC401fw:b-gg3w1u) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/README#L4)):*
> ```
> ================
> 
> Just run `make regression`.
> ```

Please add "in LibOS/shim/test/regression directory"
___
*[ltp/runltp_xml.py, line 17 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_FeA2-j0JZW7t76rh:-Ll_FeA2-j0JZW7t76ri:buucz02) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L17)):*
> ```Python
> #
> # You should have received a copy of the GNU General Public License
> # along with this program.  If not, see <http://www.gnu.org/licenses/>.
> ```

We use LGPL 3.0, please use it here.
___
*[ltp/runltp_xml.py, line 63 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_I2OHBA-DRnCwwNq1:-Ll_I2OHBA-DRnCwwNq2:b-l6uipj) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L63)):*
> ```Python
>     # ConfigParser raises NoSectionError for nonexistent sections, even when the
>     # option value is set in [DEFAULT]. That's documented and expected, but not
>     # what we want. We wan't to return the default value even for nonexistent
> ```

`wan't` -> `want`


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 13 of 15 files reviewed, 6 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow and @yamahata)

___
*[ltp/.gitignore, line 4 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LlYShJ_0XYyE51yHjo5:-LlbKQvhARRChSOu0kE_:b-jp377f) ([raw file](https://github.com/oscarlab/graphene-tests/blob/4a5426defa82a2d36c191c33f01dc83c8ea50455/ltp/.gitignore#L4)):*
<details><summary><i>Previously, yamahata wrote…</i></summary><blockquote>

> ```
> /Broken.csv
> /Failed.csv
> /Passed.csv
> ```

Because this PR removes block_tests.awk etc, those files won't be created any more, correct?
</blockquote></details>

correct
___
*[ltp/ltp.conf, line 441 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_D5fVB4z09G7gg0DE:-LlbKZ3C445ejkRBXvhg:b2nhky0) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/ltp.conf#L441)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Can we name this file `ltp_conf.ini` instead of using modelines? Using a proper extension would be more portable.
</blockquote></details>

No.
___
*[ltp/README, line 4 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_DpxeEAR6irC401fv:-LlbKeRK3xHQzinqa8qL:bixm5gb) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/README#L4)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Please add "in LibOS/shim/test/regression directory"
</blockquote></details>

Nope, in current directory. This is how it was run before this MR, which is unfortunate (it really should be named `make check`).
___
*[ltp/runltp_xml.py, line 17 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_FeA2-j0JZW7t76rh:-LlbdANb4emAz8y1T_Nj:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L17)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

We use LGPL 3.0, please use it here.
</blockquote></details>

Done.
___
*[ltp/runltp_xml.py, line 63 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_I2OHBA-DRnCwwNq1:-Llbd4qh9he6ZioF-3zm:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L63)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`wan't` -> `want`
</blockquote></details>

Done.
___
*[ltp/runltp_xml.py, line 71 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LlYW0oV3GtwIxntSQ5M:-LlbKpau-qkm7Qd4MEjP:b7abpf0) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L71)):*
<details><summary><i>Previously, yamahata wrote…</i></summary><blockquote>

Oh you wrote your own test runner and result formatter.
Isn't there any suitable existing one?
Just I'm curious.  This implementation is not so long, so I'm fine with this.
</blockquote></details>

All the existing runners I know of rely on a priori knowledge of all tests before running them, but we don't know the list before executing the binaries. Using parts of runners (for example only XML generators) would be tricky, as those are typically private, unstable APIs. And we'd need config file anyway.


<!-- Sent from Reviewable.io -->

dimakuv reviewed (dismissed):
> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 8 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow and @woju)

___
*[ltp/Makefile, line 45 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliEs2i9u7-OWC_gYZ3:-LliEs2i9u7-OWC_gYZ4:bw3iets) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/Makefile#L45)):*
> ```Makefile
> 
> ltp-sgx.xml: $(BUILDDIR)/runtest/syscalls $(target)
> 	./runltp_xml.py -o sgx=1 $< > $@
> ```

Can we call it in `regression` target when `SGX_RUN=1` envvar is set? (This is how we used to run tests on local machines.) I.e. `make regression SGX_RUN=1` should be an alias to `ltp-sgx.xml`.
___
*[ltp/README, line 4 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_DpxeEAR6irC401fv:-LliFM6V1bpHLL2KdvQC:b-t09nn4) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/README#L4)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Nope, in current directory. This is how it was run before this MR, which is unfortunate (it really should be named `make check`).
</blockquote></details>

Yes, it should be run in current directory. That step (2) in the original README was incorrect.
___
*[ltp/README, line 24 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliFceh8oEbfAnPMD4N:-LliFceh8oEbfAnPMD4O:b-1pytfg) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L24)):*
> ```
> 
> Global options:
> - `jobs`: run that many tests in parallel (default is number of CPUs)
> ```

For the `make regression SGX_RUN=1` case, we must force jobs to be 1. SGX doesn't do well with parallel tests. Not sure how to better implement this logic (hardcore in the Python script? and update this README?).
___
*[ltp/README, line 30 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliG1gUC_gPlYSfwUgc:-LliG1gUC_gPlYSfwUgd:bq3xx8h) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L30)):*
> ```
> 
> Per-binary options:
> - `skip`: if true-ish, do not attempt to run the binary (default: false).
> ```

`true-ish`? Why not `true`?
___
*[ltp/README, line 76 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliGJK31C6nTGPzpoxZ:-LliGJK31C6nTGPzpox_:b-5zke8k) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L76)):*
> ```
> recv01,1
> recv01,2
> ```

Please remove one empty line


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 12 of 18 files reviewed, 8 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv, @mkow, and @yamahata)

___
*[ltp/Makefile, line 45 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliEs2i9u7-OWC_gYZ3:-Llq1LOT2963Ww_7g8Sj:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/Makefile#L45)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we call it in `regression` target when `SGX_RUN=1` envvar is set? (This is how we used to run tests on local machines.) I.e. `make regression SGX_RUN=1` should be an alias to `ltp-sgx.xml`.
</blockquote></details>

Done.
___
*[ltp/README, line 24 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliFceh8oEbfAnPMD4N:-LmkOuvA9Dsko4hslzMD:bkfdgpg) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L24)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

For the `make regression SGX_RUN=1` case, we must force jobs to be 1. SGX doesn't do well with parallel tests. Not sure how to better implement this logic (hardcore in the Python script? and update this README?).
</blockquote></details>

This was done as a default, to allow the user to override this anyway.
___
*[ltp/README, line 30 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliG1gUC_gPlYSfwUgc:-Llq1encCD7XWaWm6j0J:bt91gmj) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L30)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`true-ish`? Why not `true`?
</blockquote></details>

https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean
___
*[ltp/README, line 76 at r3](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LliGJK31C6nTGPzpoxZ:-Llq1m9e3NYboQdRMkUF:bofokfn) ([raw file](https://github.com/oscarlab/graphene-tests/blob/d3f6c21288fd2b5db886f90dad114fb2208930f6/ltp/README#L76)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please remove one empty line
</blockquote></details>

This was concatenated verbatim from FLAKY I think.


<!-- Sent from Reviewable.io -->

mkow reviewed (changes requested):
FYI: I verified and resolved @dimakuv's comments (he's on vacation right now and asked me to take over his blocking discussions).

> Reviewed 1 of 1 files at r2, 1 of 3 files at r3, 6 of 6 files at r4.
> Dismissed @dimakuv from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-:-Lmo-tdtCTmEi8T-hnW3:b-c5bzv1)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @woju)

___
*[ltp/conflint, line 16 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnvCbu1rp7Hv-nigMi:-LmnvCbu1rp7Hv-nigMj:b2n46tm) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/conflint#L16)):*
> ```
> # GNU General Public License for more details.
> #
> # You should have received a copy of the GNU General Public License
> ```

Should be LGPL here.
___
*[ltp/conflint, line 21 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnxNZn4PyhylZfixiT:-LmnxNZn4PyhylZfixiU:b-eig56s) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/conflint#L21)):*
> ```
> 
> #
> # usage: ./conflint < ltp.conf
> ```

Could you run this in Jenkins?
___
*[ltp/ltp.conf, line 441 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_D5fVB4z09G7gg0DE:-Lmnyt8s1T5WW3bEvuEq:bb76mu4) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/ltp.conf#L441)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

No.
</blockquote></details>

Why?
___
*[ltp/ltp-sgx.conf, line 5 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnzIZI2yX-TQW-ihEQ:-LmnzIZI2yX-TQW-ihER:b9dwfp2) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/ltp-sgx.conf#L5)):*
> ```
> 
> [DEFAULT]
> timeout = 30
> ```

I'd prefer to set this to 45, SGX is slower than vanilla Linux. Basically, I think all timeouts on SGX should be ~1.5x larger than on other PALs.
___
*[ltp/ltp-sgx.conf, line 2559 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lmo-9CUCZnicMA3B59g:-Lmo-9CUCZnicMA3B59h:b-7ropln) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/ltp-sgx.conf#L2559)):*
> ```
>     17
> 
> # vim: ft=dosini
> ```

ditto
___
*[ltp/runltp_xml.py, line 17 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_FeA2-j0JZW7t76rh:-LmnvKA9CtmV07VCfwRF:b-l66b0l) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L17)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

you left GPL in the last sentence ;)


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow)

___
*[ltp/conflint, line 16 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnvCbu1rp7Hv-nigMi:-Lmo9TUv31xqSeIlVH6B:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/conflint#L16)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Should be LGPL here.
</blockquote></details>

Done.
___
*[ltp/conflint, line 21 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnxNZn4PyhylZfixiT:-Lmo9lZe0PxAS-9xnGQl:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/conflint#L21)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Could you run this in Jenkins?
</blockquote></details>

Done.
___
*[ltp/ltp.conf, line 441 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_D5fVB4z09G7gg0DE:-Lmo9nbN4nXrXZyJxEBd:b-vns0h0) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/ltp.conf#L441)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why?
</blockquote></details>

Because this is configuration file. It should be named .conf, as is customary in the Linux world.
___
*[ltp/ltp-sgx.conf, line 5 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmnzIZI2yX-TQW-ihEQ:-LmoA1Jg7xotDghZf0lu:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/ltp-sgx.conf#L5)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

I'd prefer to set this to 45, SGX is slower than vanilla Linux. Basically, I think all timeouts on SGX should be ~1.5x larger than on other PALs.
</blockquote></details>

Done.
___
*[ltp/ltp-sgx.conf, line 2559 at r4](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lmo-9CUCZnicMA3B59g:-LmoB227DAHv-2gqLpOK:b2nhktp) ([raw file](https://github.com/oscarlab/graphene-tests/blob/0636d0b29303f7b105d963e9d377b075f7ce207a/ltp/ltp-sgx.conf#L2559)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

OK.
___
*[ltp/runltp_xml.py, line 17 at r1](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Ll_FeA2-j0JZW7t76rh:-LmoA62gCK8r4pmrZ9Vl:b4j7z3d) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c60244036c9e0c864916cab7e3f0e2b5746ca785/ltp/runltp_xml.py#L17)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

you left GPL in the last sentence ;)
</blockquote></details>

Yes, I did. Corrected.


<!-- Sent from Reviewable.io -->

mkow reviewed (changes requested):
> Reviewed 5 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @woju)

___
*[ltp/conflint, line 41 at r5](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmoDBo-08pzeHzEZKX_:-LmoDBo-08pzeHzEZKXa:bdj4y51) ([raw file](https://github.com/oscarlab/graphene-tests/blob/555ae9d03ec3d579167da1a55174ce31822e70df/ltp/conflint#L41)):*
> ```
> 
> if __name__ == '__main__':
>     sys.exit(main())
> ```

Hmm, if someone makes 256 mistakes then it will result in a success ;)


<!-- Sent from Reviewable.io -->

mkow reviewed (dismissed):
> Reviewed 1 of 2 files at r6, 2 of 2 files at r7.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)



<!-- Sent from Reviewable.io -->

mkow reviewed (dismissed):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)



<!-- Sent from Reviewable.io -->

mkow reviewed (dismissed):
> Reviewed 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)



<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 15 of 18 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow)

___
*[ltp/conflint, line 41 at r5](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LmoDBo-08pzeHzEZKX_:-Lmoj65GDg0D6fuGe5rH:bsj646i) ([raw file](https://github.com/oscarlab/graphene-tests/blob/555ae9d03ec3d579167da1a55174ce31822e70df/ltp/conflint#L41)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Hmm, if someone makes 256 mistakes then it will result in a success ;)
</blockquote></details>

Corrected.


<!-- Sent from Reviewable.io -->

yamahata reviewed (dismissed):
> Reviewed 1 of 3 files at r3, 8 of 8 files at r9.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @woju)

___
*[ltp/contrib/conf_lint.py, line 36 at r9](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LnOfd0Q147KuStzXKUW:-LnOfd0Q147KuStzXKUX:b-24odsq) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c988312436747440aeed255c6f5ead57031ab562/ltp/contrib/conf_lint.py#L36)):*
> ```Python
>     with file:
>         prev = ''
>         for i, line in enumerate(file):
> ```

Doesn't configparser provide order?
I thought, default dict_type  is collections.OrderedList.
___
*[ltp/contrib/conf_lint.py, line 41 at r9](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LnOfqL-AhOQKGdyOZ1p:-LnOfqL-AhOQKGdyOZ1q:b-h04311) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c988312436747440aeed255c6f5ead57031ab562/ltp/contrib/conf_lint.py#L41)):*
> ```Python
>                 continue
>             line = line.strip(' []')
>             if line < prev:
> ```

I like enforcement of the order.


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @yamahata)

___
*[ltp/contrib/conf_lint.py, line 36 at r9](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LnOfd0Q147KuStzXKUW:-LnTCTpR8rT5QrO-vSxy:b-sps0dc) ([raw file](https://github.com/oscarlab/graphene-tests/blob/c988312436747440aeed255c6f5ead57031ab562/ltp/contrib/conf_lint.py#L36)):*
<details><summary><i>Previously, yamahata wrote…</i></summary><blockquote>

Doesn't configparser provide order?
I thought, default dict_type  is collections.OrderedList.
</blockquote></details>

Maybe it does, but using configparser API there is no way to know, which line happens to contain the error. That's the reason for custom parser.


<!-- Sent from Reviewable.io -->

dimakuv reviewed (changes requested):
> Reviewed 1 of 1 files at r10.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @woju)

___
*[ltp/ltp.conf, line 1163 at r10](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lo6IBNNBI_4QT_IBNSu:-Lo6IBNNBI_4QT_IBNSv:bc7fxib) ([raw file](https://github.com/oscarlab/graphene-tests/blob/25f27a74cd4eb8b022b4ee9e14a10a731491d7a0/ltp/ltp.conf#L1163)):*
> ```
> 
> [fstatat01]
> must-pass =
> ```

Can we add tests 1 and 3 here? That way, this PR can also close https://github.com/oscarlab/graphene-tests/pull/7. Please also update the description of this PR on GitHub that it "Closes #7".


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 18 of 20 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @yamahata)

___
*[ltp/ltp.conf, line 1163 at r10](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lo6IBNNBI_4QT_IBNSu:-LoQIbol8SCQhpyG5Q_j:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/25f27a74cd4eb8b022b4ee9e14a10a731491d7a0/ltp/ltp.conf#L1163)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we add tests 1 and 3 here? That way, this PR can also close https://github.com/oscarlab/graphene-tests/pull/7. Please also update the description of this PR on GitHub that it "Closes #7".
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

dimakuv reviewed (dismissed):
> Reviewed 2 of 2 files at r11.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel)



<!-- Sent from Reviewable.io -->

woju committed:
[LTP] temporary workaround for EISDIR problem

oscarlab/graphene#1023

woju committed:
[LTP] Temporarily disable lseek02

The test is failing for unknown, unrelated reason and blocks merge.

oscarlab/graphene#1027

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 11 of 23 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @mkow, and @yamahata)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqadgNDCuOUw7I0rrZN:-LqadgNDCuOUw7I0rrZO:b-ytr3m1) (no related file):*
TODO: port #46

---


<!-- Sent from Reviewable.io -->

woju committed:
[LTP] temporary workaround for mmap EACCESS problem

oscarlab/graphene#1075

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 11 of 25 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @mkow, and @yamahata)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqadgNDCuOUw7I0rrZN:-LqlP6HMDO1pOgayBU-r:b-896fix) (no related file):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

TODO: port #46
</blockquote></details>

Done.

---


<!-- Sent from Reviewable.io -->

dimakuv reviewed (changes requested):
> Reviewed 10 of 15 files at r1, 7 of 13 files at r12, 9 of 9 files at r13.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, 10 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @woju)

___
*[ltp/Makefile, line 22 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqvye-66jA7P1i6gA37:-Lqvye-66jA7P1i6gA38:b-lqa78y) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L22)):*
> ```Makefile
> 
> $(TESTCASEDIR)/pal_loader: $(BUILDDIR)/runltp
> 	ln -sf $(call relative-to,$(dir $@),../../../Runtime/pal_loader) $@
> ```

Can we replace this with a variable `GRAPHENE_PALLOADER` from `makevars.mk`?
___
*[ltp/Makefile, line 31 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvyCH9BY61XHkLty-o:-LqvyCH9BY61XHkLty-p:b-3y5bbp) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L31)):*
> ```Makefile
> 
> $(TESTCASEDIR)/Makefile: Makefile.testcases
> 	ln -sf ../../../../$< $@
> ```

Can we replace `../../../../` with a variable `RELATIVEROOTDIR` (or some better name) from `makevars.mk`? These slashes are confusing here.
___
*[ltp/Makefile, line 39 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvxL_A3BWNsPD8q-yu:-LqvxL_A3BWNsPD8q-yv:b6fa5yk) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L39)):*
> ```Makefile
> .PHONY: regression
> regression:
> 	if test "$(SGX_RUN)" -eq 1; then \
> ```

`SGX_RUN` was deprecated, it is simply `SGX` now
___
*[ltp/Makefile.testcases, line 1 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvyrW-2Kw4509d_RNl:-LqvyrW-2Kw4509d_RNm:b-lsf10o) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile.testcases#L1)):*
> ```
> include ../../../../makevars.mk
> ```

Can we replace `../../../../` with a variable `RELATIVEROOTDIR` (or some better name) from `makevars.mk`? These slashes are confusing here. Here and below in this file.
___
*[ltp/makevars.mk, line 3 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqvyup99KEUx5nihvWJ:-Lqvyup99KEUx5nihvWK:b-fiioqs) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/makevars.mk#L3)):*
> ```Makefile
> SRCDIR = src
> BUILDDIR = opt/ltp
> TESTCASEDIR = $(BUILDDIR)/testcases/bin
> ```

Can we add:
- `RELATIVEROOTDIR = ../../../../  # LTP root relative to opt/ltp/testcases/bin`
- `GRAPHENE_PALLOADER = $(call relative-to,$(dir $@),../../../Runtime/pal_loader)`
___
*[ltp/README.rst, line 4 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvzRKE3c7AYLGqOS7U:-LqvzRKE3c7AYLGqOS7V:b-b0jqi1) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L4)):*
> ```reStructuredText
> ================
> 
> Just run ``make regression``.
> ```

`Just run make ltp.xml or make ltp-sgx.xml (or they alias make regression)`
___
*[ltp/README.rst, line 6 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvzZbaAnhsLP9GZBH0:-LqvzZbaAnhsLP9GZBH1:be8i1df) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L6)):*
> ```reStructuredText
> Just run ``make regression``.
> 
> Test results are reported as an XML file, ``ltp.xml``, which is for consumption
> ```

`ltp.xml / ltp-sgx.xml`
___
*[ltp/README.rst, line 54 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw-NoU7SL-QqZBu4ZH:-Lqw-NoU7SL-QqZBu4ZI:bpudoxw) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L54)):*
> ```reStructuredText
> -----------
> 
> These should be debugged, or revisited after memory debugging PRs are merged.
> ```

Just remove this sentence, it's obvious.
___
*[ltp/README.rst, line 61 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw04Z82nWZFgWvJDNF:-Lqw04Z9AdHvvGrpjN1q:b241m0d) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L61)):*
> ```reStructuredText
>    process sharing its parents address space. Graphene doesn't support this exotic model. Bug exposed by #1034.
> 
>    waitpid03,1 and 2 - fails about 20% of the time
> ```

Starting from here, this is really old stuff. Most of it was fixed already (on Linux PAL), so I suggest to delete all these lines from waitpid03 and up to kill12 (kill12 must stay).
___
*[ltp/README.rst, line 100 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw0I5Q0LKSS5aHZ1du:-Lqw0I5RDXnvossyUB6N:blq201g) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L100)):*
> ```reStructuredText
>    Intermittent failure on Linux debug host
>    recvmsg01,1
>    recvmsg01,2
> ```

Ending here, this is really old stuff. Most of it was fixed already (on Linux PAL), so I suggest to delete all these lines from waitpid03 and up to kill12 (kill12 must stay).


<!-- Sent from Reviewable.io -->

woju reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: 23 of 27 files reviewed, 10 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv)

___
*[ltp/Makefile, line 22 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqvye-66jA7P1i6gA37:-Lqw_C9K-RVl-z937P6L:bsguy6j) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L22)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we replace this with a variable `GRAPHENE_PALLOADER` from `makevars.mk`?
</blockquote></details>

What for? I think this is the only place this is used.
___
*[ltp/Makefile, line 31 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvyCH9BY61XHkLty-o:-Lqw_Gb6-urw67TBm9x4:bi2ge) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L31)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we replace `../../../../` with a variable `RELATIVEROOTDIR` (or some better name) from `makevars.mk`? These slashes are confusing here.
</blockquote></details>

Is `$(abspath $<)` OK?
___
*[ltp/Makefile, line 39 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvxL_A3BWNsPD8q-yu:-LqwWT2GB6REick-WWm7:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile#L39)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`SGX_RUN` was deprecated, it is simply `SGX` now
</blockquote></details>

Done.
___
*[ltp/Makefile.testcases, line 1 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvyrW-2Kw4509d_RNl:-LqwTN8Z7mw8ZyJ89M2B:b-gp5mub) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/Makefile.testcases#L1)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we replace `../../../../` with a variable `RELATIVEROOTDIR` (or some better name) from `makevars.mk`? These slashes are confusing here. Here and below in this file.
</blockquote></details>

Yes, everywhere apart from this particular line, because that variable would not be available before `include`.
___
*[ltp/makevars.mk, line 3 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqvyup99KEUx5nihvWJ:-LrA4AoD59fkAiiJDepQ:bjseo3r) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/makevars.mk#L3)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can we add:
- `RELATIVEROOTDIR = ../../../../  # LTP root relative to opt/ltp/testcases/bin`
- `GRAPHENE_PALLOADER = $(call relative-to,$(dir $@),../../../Runtime/pal_loader)`
</blockquote></details>

ROOTDIR done, wrt GRAPHENE_PALLOADER see comment for `Makefile`.
___
*[ltp/README.rst, line 4 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvzRKE3c7AYLGqOS7U:-LqwV4pU6X0X9roFKonJ:b-xl3y03) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L4)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`Just run make ltp.xml or make ltp-sgx.xml (or they alias make regression)`
</blockquote></details>

`make regression` is an alias to make `ltp[-sgx].xml`, depending on `SGX` variable.
___
*[ltp/README.rst, line 6 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-LqvzZbaAnhsLP9GZBH0:-LrA3wMQ3wMZQBvglgOl:b-896fix) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L6)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`ltp.xml / ltp-sgx.xml`
</blockquote></details>

Done.
___
*[ltp/README.rst, line 54 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw-NoU7SL-QqZBu4ZH:-LqwVY3Z1buHZomWNhkr:b-jzsay7) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L54)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Just remove this sentence, it's obvious.
</blockquote></details>

Removed. This was carried from the previous `FLAKY` file, but is not important.
___
*[ltp/README.rst, line 61 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw04Z82nWZFgWvJDNF:-LqwVhWs5GJx4jeDLCph:b-7y05uo) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L61)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Starting from here, this is really old stuff. Most of it was fixed already (on Linux PAL), so I suggest to delete all these lines from waitpid03 and up to kill12 (kill12 must stay).
</blockquote></details>

I don't know, was this reflected in `FLAKY`? I'd like this PR to be as copy-paste as possible WRT to non-essential content. That essential content is already much.
___
*[ltp/README.rst, line 100 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw0I5Q0LKSS5aHZ1du:-LrA44AG3JbyqYSYlIxq:b-py0s5p) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L100)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Ending here, this is really old stuff. Most of it was fixed already (on Linux PAL), so I suggest to delete all these lines from waitpid03 and up to kill12 (kill12 must stay).
</blockquote></details>

ACK, see previous comment.


<!-- Sent from Reviewable.io -->

dimakuv reviewed (dismissed):
> Reviewed 4 of 4 files at r14.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

___
*[ltp/README.rst, line 61 at r13](https://reviewable.io/reviews/oscarlab/graphene-tests/21#-Lqw04Z82nWZFgWvJDNF:-LrAVTa4EHP8aIGzk84g:bf9my35) ([raw file](https://github.com/oscarlab/graphene-tests/blob/ad8ebd3d46f963a2793cf2be64781e0caec30e26/ltp/README.rst#L61)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

I don't know, was this reflected in `FLAKY`? I'd like this PR to be as copy-paste as possible WRT to non-essential content. That essential content is already much.
</blockquote></details>

OK, we'll clean this later.


<!-- Sent from Reviewable.io -->

yamahata reviewed (dismissed):
> Reviewed 7 of 13 files at r12, 5 of 9 files at r13, 4 of 4 files at r14.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from different teams (1 more required, approved so far: Intel)



<!-- Sent from Reviewable.io -->

omeg reviewed (dismissed):
> Reviewed 10 of 15 files at r1, 1 of 3 files at r3, 7 of 13 files at r12, 5 of 9 files at r13, 4 of 4 files at r14.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

omeg reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)



<!-- Sent from Reviewable.io -->

dimakuv reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene-tests/21)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
